recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -123,15 +123,19 @@ this.NormandyDriver = function(sandboxManager) {
       return ret;
     },
 
-    createStorage(keyPrefix) {
-      let storage;
-      try {
-        storage = Storage.makeStorage(keyPrefix, sandbox);
-      } catch (e) {
-        log.error(e.stack);
-        throw e;
+    createStorage(prefix) {
+      const storage = new Storage(prefix);
+
+      // Wrapped methods that we expose to the sandbox. These are documented in
+      // the driver spec in docs/dev/driver.rst.
+      for (const method of ["getItem", "setItem", "removeItem", "clear"]) {
+        storage[method] = sandboxManager.wrapAsync(storage[method], {
+          cloneArguments: true,
+          cloneInto: true,
+        });
       }
-      return storage;
+
+      return sandboxManager.cloneInto(storage, {cloneFunctions: true});
```

Comment:
Instead of modifying and passing the storage object directly, could we make an interface object to pass in?

```js
const storage  = new Storage(prefix);

const interface = {};
for (const method of ["getItem", "setItem", "removeItem", "clear"]) {
  interface[method] = sandboxManager.wrapAsync(storage[method], {
    cloneArguments: true,
    cloneInto: true,
  });
}

return sandboxManager.cloneInto(interface, {cloneFunctions: true});
```

I think this will help avoid exposing unintended functions to the sandbox. I also think that it would be better to not modify the storage object, so any `this.foo` method calls it makes will use unmodified versions of the methods. Would that work?